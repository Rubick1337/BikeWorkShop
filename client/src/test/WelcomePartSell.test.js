// WelcomePartSell.test.js
import React from 'react';
import { render } from '@testing-library/react';
import WelcomPartSell from '../components/WelcomePartSell/WelcomePartSell'; // Убедитесь, что путь правильный

describe('WelcomPartSell Component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<WelcomPartSell />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders with correct class names and text content', () => {
        const { container } = render(<WelcomPartSell />);

        // Проверка наличия контейнера с фоном
        const backgroundPart = container.querySelector('.background__sell__part');
        expect(backgroundPart).toBeInTheDocument();

        // Проверка заголовка h2
        const title = container.querySelector('h2');
        expect(title).toBeInTheDocument();
        expect(title).toHaveTextContent('Ride Free, Ride Far');

        // Проверка заголовка h3
        const subtitle = container.querySelector('h3');
        expect(subtitle).toBeInTheDocument();
        expect(subtitle).toHaveTextContent('Посмотрите нашу коллекцию');

        // Проверка ссылки
        const link = container.querySelector('.link__sell');
        expect(link).toBeInTheDocument();
        expect(link).toHaveTextContent('За покупками');
    });
});