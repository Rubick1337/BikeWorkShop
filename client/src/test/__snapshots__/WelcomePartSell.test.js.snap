// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomPartSell Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container__sell__bike"
  >
    <div
      class="background__sell__part"
    >
      <div
        class="container__text__sell"
      >
        <h2>
          Ride Free, Ride Far
        </h2>
        <h3>
          Посмотрите нашу коллекцию
        </h3>
        <a
          class="link__sell"
          href="#sellPart"
        >
          За покупками
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
